xds/src/test/java/io/grpc/xds/XdsTestControlPlaneService.java

```diff
@@ -135,12 +135,15 @@ public void run() {
                   new Object[]{value.getResourceNamesList(), value.getErrorDetail()});
               return;
             }
+
             String resourceType = value.getTypeUrl();
-            if (!value.getResponseNonce().isEmpty()
-                && !String.valueOf(xdsNonces.get(resourceType)).equals(value.getResponseNonce())) {
+            if (!value.getResponseNonce().isEmpty() && xdsNonces.containsKey(resourceType)
```

Comment:
this change also seems not related?

Reply:
This fix is needed to support anything trying a nonce other than the initial empty value.

Reply:
I still don't see this change related to "support multiple xds client". Looks this file can be reverted without impacting anything?
But I see this potentially avoid NPE when it receives another resource type other than the hardcoded resource type in `xdsNonces` keyset.